libs/plugin/src/generators/convert-di-to-inject/generator.ts

```diff
@@ -0,0 +1,244 @@
+					decoratorDecl.getName(),
+				);
+			});
+			if (!applicableDecorator) continue;
+
+			const convertedDeps = new Set<string>();
+
+			targetClass.getConstructors().forEach((constructor) => {
+				constructor.getParameters().forEach((param, index) => {
+					const { name, type, decorators, scope, isReadonly } =
+						param.getStructure();
+
+					let shouldUseType = false;
+					let toBeInjected = type; // default to type
+					const flags = [];
+
+					if (decorators.length > 0) {
+						decorators.forEach((decorator) => {
+							if (decorator.name === 'Inject') {
+								toBeInjected = decorator.arguments[0]; // use the argument of the @Inject decorator
+								if (toBeInjected !== type) {
+									shouldUseType = true;
+								}
+							}
+							if (decorator.name === 'Optional') {
+								flags.push('optional');
+							}
+							if (decorator.name === 'Self') {
+								flags.push('self');
+							}
+							if (decorator.name === 'SkipSelf') {
+								flags.push('skipSelf');
+							}
+							if (decorator.name === 'Host') {
+								flags.push('host');
+							}
+						});
+					}
+
+					// if type is (ElementRef or TemplateRef) or should use type, add it as inject generic
+
+					let injection = 'inject';
+
+					const typeHasGenerics = type.toString().includes('<');
+
+					if (shouldUseType || typeHasGenerics) {
+						injection += `<${type}>`;
+					}
+
+					targetClass.insertProperty(index, {
+						name,
+						initializer: `${injection}(${toBeInjected}${flags.length > 0 ? `, { ${flags.map((flag) => flag + ': true').join(', ')} }` : ''})`,
+						scope,
+						isReadonly,
+						leadingTrivia: '  ',
+					});
+
+					convertedDeps.add(name);
+
+					// check if service was used inside the constructor without 'this.' prefix
+					// if so, add 'this.' prefix to the service
+					constructor.getStatements().forEach((statement) => {
+						if (Node.isExpressionStatement(statement)) {
+							const expression = statement.getExpression();
+							if (Node.isCallExpression(expression)) {
+								const expressionText = expression.getText();
+								if (
+									expressionText.includes(name.toString()) &&
+									!expressionText.includes(`this.${name.toString()}`)
+								) {
+									const newExpression = expressionText.replace(
+										name.toString(),
+										`this.${name}`,
+									);
+									statement.replaceWithText(newExpression);
+								}
+							}
+						}
+					});
+				});
+
+				if (convertedDeps.size > 0 && !hasInjectImport) {
+					sourceFile.insertImportDeclaration(0, {
+						namedImports: ['inject'],
+						moduleSpecifier: '@angular/core',
+						leadingTrivia: '  ',
+					});
+				}
+
+				constructor.getParameters().forEach((param) => {
+					if (convertedDeps.has(param.getName())) {
+						param.remove();
+					}
+				});
+
+				if (
+					constructor.getParameters().length === 0 &&
+					constructor.getBodyText().trim() === ''
```

Comment:
@eneajaho  I think you should check if the constructor has `super()` only, if that's the case the constructor should be removed as well 
```
constructor.getBodyText().trim() === 'super()'
```

Reply:
I think we should check first if the schematic converts also the DI injections in the plain classes - ones being decorated anyway. if yes, then we can also remove the constructors with super() only. I guess 👀